### PR TITLE
Adds ravens 2

### PR DESCRIPTION
**Pull request by Raven567**
Ravenflyfront
RavenRightflyt
Ravenright4
Ravendead

## What Does This PR Do

It adds the raven mob to the game.
## Why It's Good For The Game

It adds a wonderful companion/competitor to poly, and as the old saying goes, monopolies are bad. Poly is an outstanding pet that stands above them all, and I want a pet to be its equal. Also the goons have a very large variety of birds compared to us here at paradise. And it is a draw for some insane people like me to know that there are birds in the game ha ha. I think of space station 13 not really as a space game, but more as a game of infinite potential, its like playing dnd, but you can play all of the time.
## Images of changes

image

## Testing

image

I compiled and tested each of the abilities, 
## Changelog
:cl:
Adds ravens to the game.
/:cl:

### CODE REVIEW

**Comment by Raven567** [comment]
It would be nice for ravens to be accessible through more means than just xeno biology, Preferably id like to see them as a pet, like runtime for one of the department heads or the librarian or something. We can not currently add them to crates sadly, as I am restrict due to the economics refactor.  Please leave me your thoughts and suggestions, for now this pr only adds the ravens, and they will only be accessible to admins, and xenobiology. 

Also thank you all for your feed back, I know I get a little salted time to time! It has definitely  been invaluable, extra specially the feed back that's made me grind me teeth. Its ultimately lead to higher quality.

**Comment by Necaladun** [comment]
Hi there,

I'll be objecting to this as general bloat. Everyone has favorite animals, and if we start allowing everyone to add there's it's just going to be ridiculous. We already have absolutely heaps.

This is a game about a space station and general chaos and betrayel on it - not about having  your favorite animal as a pet. Was there perhaps some other demand or want for this I would be open to it, but from your username, profile pic, and general behavior, this seems an entirely self-serving PR that only benefits you.

**Comment by Raven567** [comment]
> Hi there,
> 
> I'll be objecting to this as general bloat. Everyone has favorite animals, and if we start allowing everyone to add there's it's just going to be ridiculous. We already have absolutely heaps.
> 
> This is a game about a space station and general chaos and betrayel on it - not about having your favorite animal as a pet. Was there perhaps some other demand or want for this I would be open to it, but from your username, profile pic, and general behavior, this seems an entirely self-serving PR that only benefits you.

Perhaps it is, but ravens are cool, and you are correct this means alot to me. I have put alot of passion into it. I am hoping that others can enjoy this as well, and have another bird really can't hurt?

**Review by Edan52 — CHANGES_REQUESTED** [review]

**Comment by PacifistDalek** [comment]
Hello Raven. It sounds like you're having a rough time, and I'm sorry about that. I want you to know that many of us appreciate your enthusiasm and work on this, however the nature of SS13 as a project has the downside that sometimes things you get enthusiastic about don't fit with the direction the staff have in mind. All of us who create PRs here have had some of ours that we're very passionate about get rejected or closed. It isn't fun, but it's the nature of the beast (one of mine, for instance was #16882, porting Eris Slimes).

On the flipside, one advantage of creating PRs for SS13 is that you can very easily join another server's community and see if they're interested, or even make your own fork of a server's code and run your own server where you get to call the shots!

Creative outlets are a fantastic way to help you feel better about yourself and improve your life (it has absolutely been a huge help for me!). If you're having a hard time dealing with the challenges of contributing to a collaborative project like our server, you may want to consider finding a different creative outlet (more personally creative, where you get to have last call on things, and you can build up your skills, your confidence, and your experience, so you're able to have a more enjoyable time in the future working on collaborative projects if things don't go the way you hope)

Of course, if you want to keep going here, please do! Keep trying different ideas and working on them with the passion you have been :) Just know that collaborative creative projects do have their challenges, and don't be worried if you have to build up to dealing with those challenges!

File: code/modules/mob/living/simple_animal/friendly/raven.dm [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	melee_damage_upper = 2
+	response_help  = "pets"
+	response_disarm = "shoos"
+	response_harm   = "kicks"
+	ventcrawler = VENTCRAWLER_ALWAYS
+	density = FALSE
+	pass_flags = PASSTABLE | PASSMOB
+	can_hide = TRUE
+	butcher_results = list(/obj/item/reagent_containers/food/snacks/meat = 2)
+	can_collar = TRUE
```
[the remark this answers is not in the document]
Comment (Raven567) [reply code/modules/mob/living/simple_animal/friendly/raven.dm:24]: > Hello Raven. It sounds like you're having a rough time, and I'm sorry about that. I want you to know that many of us appreciate your enthusiasm and work on this, however the nature of SS13 as a project has the downside that sometimes things you get enthusiastic about don't fit with the direction the staff have in mind. All of us who create PRs here have had some of ours that we're very passionate about get rejected or closed. It isn't fun, but it's the nature of the beast (one of mine, for instance was #16882, porting Eris Slimes).
> 
> On the flipside, one advantage of creating PRs for SS13 is that you can very easily join another server's community and see if they're interested, or even make your own fork of a server's code and run your own server where you get to call the shots!
> 
> Creative outlets are a fantastic way to help you feel better about yourself and improve your life (it has absolutely been a huge help for me!). If you're having a hard time dealing with the challenges of contributing to a collaborative project like our server, you may want to consider finding a different creative outlet (more personally creative, where you get to have last call on things, and you can build up your skills, your confidence, and your experience, so you're able to have a more enjoyable time in the future working on collaborative projects if things don't go the way you hope)
> 
> Of course, if you want to keep going here, please do! Keep trying different ideas and working on them with the passion you have been :) Just know that collaborative creative projects do have their challenges, and don't be worried if you have to build up to dealing with those challenges!

This is very kind of you to write this, and I really truly appreciate it.

File: code/modules/mob/living/simple_animal/friendly/raven.dm [hunk 2]
```diff
@@ -0,0 +1,56 @@
+///the perfectly normal raven.
+/mob/living/simple_animal/pet/raven
+	name = "raven"
+	desc = "Quoth the raven nevermore! Just now in space!"
+	icon = 'icons/mob/animal.dmi'
```
[the remark this answers is not in the document]
Comment (Raven567) [reply code/modules/mob/living/simple_animal/friendly/raven.dm:5]: For now I removed it for simplification.

File: code/modules/mob/living/simple_animal/friendly/raven.dm [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+		icon_state = "raven_fly"//The ravens flying animation, Later in my refactor I intend on making this a varible for perch()
+	else
+		icon_state = "raven"
+
+/mob/living/simple_animal/pet/raven/verb/caw()//The raven caw ability button
+	set name = "Caw!"
+	set category = "Raven"
+	set desc = "caws"
+
+	if(caw_cooldown < world.time) // A caw_cooldown, to stop people being jerks
```
[the remark this answers is not in the document]
Comment (Raven567) [reply code/modules/mob/living/simple_animal/friendly/raven.dm:54]: I made some adjustments too it, I noticed one comment was even an artifact XD.
image

**Comment by Raven567** [comment]
> Hello Raven. It sounds like you're having a rough time, and I'm sorry about that. I want you to know that many of us appreciate your enthusiasm and work on this, however the nature of SS13 as a project has the downside that sometimes things you get enthusiastic about don't fit with the direction the staff have in mind. All of us who create PRs here have had some of ours that we're very passionate about get rejected or closed. It isn't fun, but it's the nature of the beast (one of mine, for instance was #16882, porting Eris Slimes).
> 
> On the flipside, one advantage of creating PRs for SS13 is that you can very easily join another server's community and see if they're interested, or even make your own fork of a server's code and run your own server where you get to call the shots!
> 
> Creative outlets are a fantastic way to help you feel better about yourself and improve your life (it has absolutely been a huge help for me!). If you're having a hard time dealing with the challenges of contributing to a collaborative project like our server, you may want to consider finding a different creative outlet (more personally creative, where you get to have last call on things, and you can build up your skills, your confidence, and your experience, so you're able to have a more enjoyable time in the future working on collaborative projects if things don't go the way you hope)
> 
> Of course, if you want to keep going here, please do! Keep trying different ideas and working on them with the passion you have been :) Just know that collaborative creative projects do have their challenges, and don't be worried if you have to build up to dealing with those challenges!

I really truly appreciate this, this was really written out of empathy and compassion, thank you. It certainly made my day reading this man.

File: code/modules/mob/living/simple_animal/friendly/raven.dm (hunk 1 above)
[the remark this answers is not in the document]
Comment (Edan52) [reply code/modules/mob/living/simple_animal/friendly/raven.dm]: On `simple_animal/friendly` yes

File: code/modules/mob/living/simple_animal/friendly/raven.dm [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+	///This varible allows you to set a cooldown on an ability, for an example usage reference the raven.dm, last line.
+	var/caw_cooldown
+
+
+/mob/living/simple_animal/pet/raven/npc_safe(mob/user)//Makes the raven a playable via the ghost menu.
+	return TRUE
+//The raven category is established here, as well as the fly and take off button
+/mob/living/simple_animal/pet/raven/verb/perch()
+	set name = "Fly/Take off!"
+	set category = "Raven"
```
Comment (Edan52) [inline code/modules/mob/living/simple_animal/friendly/raven.dm:34]: This should probably be in an existing category as new ones will cause lag.

File: code/modules/mob/living/simple_animal/friendly/raven.dm [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+	pass_flags = PASSTABLE | PASSMOB
+	butcher_results = list(/obj/item/reagent_containers/food/snacks/meat = 2)
+	gold_core_spawnable = FRIENDLY_SPAWN
+	turns_per_move = 5
+	///This varible allows you to set a cooldown on an ability, for an example usage reference the raven.dm, last line.
+	var/caw_cooldown
+
+
+/mob/living/simple_animal/pet/raven/npc_safe(mob/user)//Makes the raven a playable via the ghost menu.
+	return TRUE
```
Comment (Edan52) [inline code/modules/mob/living/simple_animal/friendly/raven.dm:30]:
```suggestion
	return TRUE
	
```

Comment (Edan52) [reply code/modules/mob/living/simple_animal/friendly/raven.dm:30]: Spacing below procs.

**Review by Edan52 — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/simple_animal/friendly/raven.dm (hunk 4 above)
Comment (Raven567) [reply code/modules/mob/living/simple_animal/friendly/raven.dm:34]: I am not entirely how to resolve this one, is there a specific file I should define it in?

Comment (Raven567) [reply code/modules/mob/living/simple_animal/friendly/raven.dm:35]: Also thank you for taking time to review my code, I thought I was on my own now.

Comment (Edan52) [reply code/modules/mob/living/simple_animal/friendly/raven.dm:35]: Yes, one of the existing verb categories, I'd probably go with `IC`

Comment (Raven567) [reply code/modules/mob/living/simple_animal/friendly/raven.dm:35]: image

File: code/modules/mob/living/simple_animal/friendly/raven.dm (hunk 1 above)
[the remark this answers is not in the document]
Comment (Raven567) [reply code/modules/mob/living/simple_animal/friendly/raven.dm]: I removed all 4 of these overwrites, so it should be good, it also seems to be testing fine as well.

**Comment by Raven567** [comment]
Also pacifists I read that, that is actually heart breaking. If its any constellation I think they looked pretty cool. 
https://github.com/ParadiseSS13/Paradise/pull/16882

**Comment by Raven567** [comment]
Small update, here is a link to the original, I finally found it.
https://github.com/ParadiseSS13/Paradise/pull/19357

**Review by Raven567 — COMMENTED** [review]
Sweet thanks for helping me with this.

**Pull request closed without merging**